File: study/lua/src/app/scene/view/activity/ActivitySubView.lua
local ViewBase = require("app.ui.ViewBase")
local ActivitySubView = class("ActivitySubView", ViewBase)

function ActivitySubView:ctor(resource)
	self._actTitle = nil
    self._isInShow = false
    ActivitySubView.super.ctor(self, resource)
end

function ActivitySubView:enterModule()
    self._isInShow = true
end

function ActivitySubView:exitModule()
    self._isInShow = false
end

function ActivitySubView:isInShow()
    return self._isInShow
end

function ActivitySubView:setTitle(str)
    self._actTitle:setTitle(str)
end

return ActivitySubView

File: study/ts/script/scene/view/activity/ActivitySubView.ts
import ViewBase from "../../ViewBase";
import CommonFullScreenActivityTitle from "../../../ui/component/CommonFullScreenActivityTitle";

const {ccclass, property} = cc._decorator;

@ccclass
export default class ActivitySubView extends ViewBase {

    @property({
        type: CommonFullScreenActivityTitle,
        visible: true
    })
    _actTitle: CommonFullScreenActivityTitle = null;

    _isInShow:boolean = false;

    enterModule() {
        this._isInShow = true;
    }
    exitModule() {
        this._isInShow = false;
    }
    isInShow() {
        return this._isInShow;
    }
    setTitle(str) {
        this._actTitle.setTitle(str);
    }

    onCreate(){

    }
    onEnter(){

    }
    onExit(){

    }
}
